airbyte-integrations/connectors/source-google-ads/source_google_ads/streams.py

```diff
@@ -186,6 +189,7 @@ def read_records(
         and update `start_date` key in the `stream_slice` with the latest read record's cursor value, then retry the sync.
         """
         while True:
+            self.logger.info("Starting a while loop iteration")
```

Comment:
how many iterations of this loop per second/minute do we do? We should make sure it's not dumping too much logs to disk. It could cause other reliability problems

Reply:
I can't say for sure. Ideally, a single iteration should be done per stream slice which should not produce too much logs I believe. Additional iterations may occur in case a page token expires.
I tried adding as little logs as possible but still to have enough information to make conclusions about what's going wrong